src/main/java/org/jruby/ext/openssl/ASN1.java

```diff
@@ -1426,6 +1458,40 @@ public IRubyObject to_der(final ThreadContext context) {
 
         byte[] toDER(final ThreadContext context) throws IOException {
             if ( isEOC() ) return new byte[] { 0x00, 0x00 };
+
+            if (isUniversal(context)) {
+                // handstitch conversion
+                final java.io.ByteArrayOutputStream out = new ByteArrayOutputStream();
+                final IRubyObject value = callMethod(context, "value");
+
+                final byte[] valueBytes;
+                if (value instanceof RubyArray) {
+                    String values = new String();
+                    final RubyArray arr = (RubyArray) value;
+                    for (final IRubyObject obj : arr.toJavaArray()) {
+                        final IRubyObject string = value.checkStringType();
+                        if (string instanceof RubyString) {
+                            values = values + string;
```

Comment:
is there a reason we do not extract the bytes directly like in the other branch (where we use `getBytes`)?

Reply:
not really, I thought that it'd be simpler to read, but I agree it's more efficient by using byte array streams. changed, feel free to reopen if there's smth missing.